### ISSUE TITLE
How does one specify environment variables?

### ISSUE THREAD

**Issue by fmr** [resin-os/resin-device-toolbox#27]
Hi all,

Apologies if the answer to this is obvious to everyone else, but I seem to be having trouble with setting environment variables using resin-os and `rdt`. Previously, I was able to set environment variables in resin.io through the dashboard, like so:

![image](https://cloud.githubusercontent.com/assets/592174/19374692/5a8ac288-9219-11e6-91ad-96247f8836c3.png)

I have started using resin-os with `rdt` to see if I can get our resin.io project to work on it. 

My setup is as follows:
- Device: raspberry pi 3
- OS on device: Resin OS
- Client: `rdt` installed via npm on OS X El Capitan

Here's the last few lines of the errors that I got when I ran `rdt push resin.local -s .`:

```
rdt push completed successfully!
* Streaming application logs..
Error: Format string '%(ENV_HOTSPOT_ENABLED)s' for 'program:hotspot.autostart' contains names ('ENV_HOTSPOT_ENABLED') which cannot be expanded. Available names: ENV_DEBIAN_FRONTEND, ENV_HOME, ENV_HOSTNAME, ENV_LC_ALL, ENV_PATH, ENV_PWD, ENV_SHLVL, ENV_TERM, ENV__, ENV_container, group_name, here, host_node_name, program_name in section 'program:hotspot' (file: '/app/deployment/supervisor.conf')
For help, use /usr/local/bin/supervisord -h
```

And here are the relevant lines (`supervisor.conf`)

```
[program:hotspot]
command=/app/hotspot/bootstrap.sh
autostart=%(ENV_HOTSPOT_ENABLED)s
# ...
```
## 
- Can someone point me in the right direction as to how to set environment variables used by the container?
- Is there any documentation about using `.resin-sync.yml` and all the available options, (especially ones specific for `local_resinos`)?

**Comment by lekkas** [comment]
Hey @fmr , 

First of all, that's a nicely filed issue 👍

The Dashboard-controlled [environment variables](http://docs.resin.io/management/env-vars/) is a runtime feature that is enabled in [resin.io](https://resin.io) provisioned devices only, so that you don't have to modify your source code.

Unlike to [resin.io](https://resin.io) provisioned devices, devices that are provisioned with [resinOS](https://resinos.io) are not connected to the resin.io cloud platform by default, so any environment variables that are required by your application will have to be specified in your source code before the application is built (i.e. before the `rdt push` command). One solution would be to add `ENV` [Dockerfile instructions](https://docs.docker.com/engine/reference/builder/#/env) to set the required variables.

**Comment by fmr** [comment]
Cheers @lekkas!

I guess I was hoping there was a way to pass environment variables without having to modify the source code, similar to how it's done with dashboard-controlled variables, or even something like `docker run -e FOO=bar ...` (or how you can pass them in `docker-compose`). I suppose I'll have to use the `ENV` instruction in the dockerfile to work around it for now.

Thanks for the feedback!

**Comment by lekkas** [comment]
@fmr right, at the moment this is one of the available workarounds but we may consider adding support for more runtime options (also check #41 which might be relevant) cc @shaunmulligan

**Comment by shaunmulligan** [comment]
@lekkas would it be difficult to add a section in `.resin-sync.yml` for env-vars? somethink like:

```
local_resinos:
  app-name: leds
  build-triggers:
    - Dockerfile: 76e9c733be2832fce5f286247e92b907d895862c243e361421787fc9b386721a
    - package.json: 4cfd9a53e41abfba755aa61f8dc44b8bb0a028a8364acb0108130b088b9fbe46
  env-vars:
    - FOO: bar
    - PROD: true
```

**Comment by lekkas** [comment]
@shaunmulligan Not difficult at all. Should we add a respective `--env` option in `rdt push` ?

**Comment by fmr** [comment]
@lekkas I second @shaunmulligan's suggestion as well as the respective `--env` option for `rdt push` :)

**Comment by lekkas** [comment]
@fmr @shaunmulligan this feature landed in Landed in `resin-device-toolbox@0.0.5`

**Comment by fmr** [comment]
That's great! Cheers @lekkas!
